src/app/blog/(posts)/godot-choosing-network-protocol/page.mdx

```diff
@@ -0,0 +1,92 @@
+import image from './opengraph-image.png';
+import imgProtocolTree from './protocol-tree.svg';
+
+export const config = {
+  author: 'nathan-flurry',
+  date: '2024-05-09',
+  title: `Choosing the Right Network Protocol for Godot Multiplayer`,
+  description: `Learn how to select the right networking protocol in Godot for optimal multiplayer gameplay, focusing on ENet, WebSocket, and WebRTC`,
+  category: 'guide',
+  keywords: [
+    'godot',
+    'multiplayer',
+    'enet',
+    'webrtc',
+    'websocket',
+    'networking-protocols',
+    'game-development',
+    'peer-to-peer'
+  ],
+  images: {
+    hero: {
+      image,
+      alt: 'hero'
+    }
+  }
+};
+
+## TL;DR
+
+Each protocol has its own tradeoff. The diagram below helps guide the decision.
+
+<Image src={imgProtocolTree} />
+
+<Info>
+  If you're just getting started with multiplayer, stick with `ENetMultiplayerPeer`. It's a safe bet and you
+  can easily change it later.
+</Info>
+
+## Choosing the Right Networking Protocol
+
+When setting up multiplayer in Godot, you need to choose the appropriate networking protocol by choosing a [MultiplayerPeer](https://docs.godotengine.org/en/stable/classes/class_multiplayerpeer.html#class-multiplayerpeer) implementation based on your game's requirements. Godot provides several options, each with its own characteristics and use cases:
+
+### `ENetMultiplayerPeer` (ENet)
+
+- Ideal for games that require fast-paced, real-time multiplayer gameplay.
+- Uses the [ENet library](http://enet.bespin.org/), which is designed for reliable, low-latency communication over UDP.
+- Supports both client-server and peer-to-peer (mesh) architectures.
+- Provides multiple channels for prioritizing different types of data.
+- Suitable for desktop and mobile platforms.
+- Pros:
+  - Low latency and reliable communication.
+  - Efficient bandwidth usage.
+  - Supports multiple channels for data prioritization.
+- Cons:
+  - Does not work with web.
+  - May be blocked on many firewalls because of UDP.
+  - Requires extra steps to [configure DTLS encryption](https://godotengine.org/article/enet-dtls-encryption/) for secure transport.
+
+For more information, see the [ENetMultiplayerPeer documentation](https://docs.godotengine.org/en/stable/classes/class_enetmultiplayerpeer.html).
+
+### `WebSocketMultiplayerPeer` (WebSocket)
+
+- Ideal for games that need to support web browsers and doesn't require peer-to-peer.
```

Comment:
The first few bullet points here talk about *where* it works, but not really *what* it works best for. I know in cons it mentions that latency will be higher than UDP, but it could be worth mentioning that it's good for certain games (say turn-based games), or that it might be limiting when playing real-time games because of TCP overhead.